src/REL/d/a/e/d_a_e_sm.cpp

```diff
@@ -13,33 +90,1372 @@ STATE_DEFINE(dAcEsm_c, Absorption);
+            mScale.set(0.8f, 0.8f, 0.8f);
+        } break;
+        case SM_LARGE: {
+            boundingBox.Set(mVec3_c(-500.f, -500.f, -500.f), mVec3_c(500.f, 750.f, 500.f));
+            mScale.set(0.4f, 0.4f, 0.4f);
+        } break;
+        case SM_MASSIVE: {
+            boundingBox.Set(mVec3_c(-800.f, -800.f, -800.f), mVec3_c(800.f, 1200.f, 800.f));
+            mScale.set(0.25f, 0.25f, 0.25f);
+        } break;
+        case SM_TINY: {
+            boundingBox.Set(mVec3_c(-150.f, -150.f, -150.f), mVec3_c(150.f, 225.f, 150.f));
+            mScale.set(1.2f, 1.2f, 1.2f);
+        } break;
+    }
+
+    updateBoundingBox();
+
+    mScale.CopyTo(mScaleTarget);
+    mScale.CopyTo(mScaleCopy2);
+
+    mObjAcch.Set(this, 1, &mAcchCir);
+    mAcchCir.SetWall(mScaleTarget.y * 100.f, mScaleTarget.x * 100.f);
+    mStts.SetRank(10);
+    mObjAcch.SetGndThinCellingOff();
+    mObjAcch.mField_0x390 = 1;
+
+    forwardMaxSpeed = -40.f;
+
+    mSph.Set(sSphSrc);
+
+    mSph.SetStts(mStts);
+
+    position.CopyTo(poscopy2);
+    position.CopyTo(mPosCopy1);
+    position.CopyTo(poscopy3);
+    poscopy3.y += 50.f;
+
+    mRotCopy.set(rotation);
+
+    field_0xB98 = 0;
+    field_0xB58 = 1.f;
+    field_0xB90 = -1;
+
+    // This is really weird...
+    if (!sSomeArrayInit) {
+        sSomeArrayInit = true;
+        bool *parr = getArray();
+        for (int i = 0; i < 8; i++) {
+            *parr++ = false;
+        }
+        *parr++ = false;
+    }
+
+    static todoStruct00::InternalData data0 = {0, 100.f, nullptr};
+    static todoStruct00::InternalData data1 = {1, 130.f, &data0};
+    field_0x9f8.Set(this, &data1, &mMdl.getModel(), 1);
+
+    mLightInfo.SetColor(mColor(0xCC, 0xFF, 0xFF, 0xFF));
+    mLightInfo.SetScale(0.f);
+
+    BlurAndPaletteManager::GetPInstance()->fn_800225F0(&mLightInfo);
+    if (field_0xBBF == 1) {
+        rotation.x = 0x8000;
+        field_0xBCC = 1;
+    }
+
+    mRotUnk.clear();
+
+    mEffArr[0].init(this);
+    mEffArr[1].init(this);
+
+    if (field_0xBBF == 1) {
+        mStateMgr.changeState(StateID_Wait);
+    } else {
+        mStateMgr.changeState(StateID_Walk);
+    }
+
+    mHealth = 100;
+
+    if (field_0xBBF != 1) {
+        forwardAccel = -3.f;
+    }
+    calcVelocity();
+    position += velocity;
+    position += mStts.GetCcMove();
+
+    mObjAcch.CrrPos(*dBgS::GetInstance());
+    fn_187_4CC0();
+
+    f32 anim_frame;
+    switch (mType) {
+        // SM_RED
+        default: {
+            targetFiTextId = 59;
+            anim_frame = 0.f;
+            mSph.SetAtModifier(AT_MOD_FIRE);
+            mType = 0;
+        } break;
+        case SM_GREEN_ALT:
+        case SM_GREEN:     {
+            targetFiTextId = 62;
+            anim_frame = 3.f;
+            mType = 1;
+        } break;
+        case SM_YELLOW_ALT:
+        case SM_YELLOW:     {
+            targetFiTextId = 61;
+            anim_frame = 2.f;
+            mSph.SetAtModifier(AT_MOD_ELECTRIC);
+            mType = 2;
```

Comment:
These three assignments could probably use the enum values

Reply:
referring to mType or targetFiTextID ?

Reply:
I meant mType here